cmd/antrea-controller/app/controller.go

```diff
@@ -93,6 +95,12 @@ const (
 	serverMinWatchTimeout = 2 * time.Hour
 )
 
+var (
+	newNodeIpamController   = nodeipam.NewNodeIpamController
+	createK8sClient         = k8s.CreateClients
+	RunAntreaControllerFunc = run
```

Comment:
I think it's hard to unit test a program's entry function as it normally involves a lot of dependencies which may need to be mocked, which is not something "unit" test is good at. 

And the point of test is not running the code once, but more about the verification, which is also why it's hard to "unit" test because running the entry code means almost all code are executed. Even antrea-controller does not many system depdendencies, there aren't many verifications can be found from the test code, not to mention antrea-agent which has a lot of system dependencies.

Overall I think it's not very worth to unit test the entry function with the price of refactoring and extra code introduced by the patch. We could test some code piece with many branches if they can be extracted to functions approriately. But I don't find many from `run` as most code just initializes modules and runs them according to configuration.

cc @antoninbas

Reply:
I also feel this way